src/App.tsx

```diff
@@ -1,14 +1,59 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(data => {
+        setTodos(data);
+      })
+      .catch(error => {
+        return (`Error while loading: ${error}`);
+      })
```

Comment:
if you return this, it will be lost, because returned value isn't used anywhere. Better throw the Error